controllers/resources.go

```diff
@@ -52,493 +53,349 @@ var (
 	mountPropagationHostToContainer = corev1.MountPropagationHostToContainer
 	mountPropagationBidirectional   = corev1.MountPropagationBidirectional
 
-	// PSP
-	pspController = func() policy.PodSecurityPolicy {
-		return policy.PodSecurityPolicy{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "lb-csi-ctrl-sa",
-			},
-			Spec: policy.PodSecurityPolicySpec{
-				FSGroup:            policy.FSGroupStrategyOptions{Rule: policy.FSGroupStrategyRunAsAny},
-				RunAsUser:          policy.RunAsUserStrategyOptions{Rule: policy.RunAsUserStrategyRunAsAny},
-				SELinux:            policy.SELinuxStrategyOptions{Rule: policy.SELinuxStrategyRunAsAny},
-				SupplementalGroups: policy.SupplementalGroupsStrategyOptions{Rule: policy.SupplementalGroupsStrategyRunAsAny},
-				Volumes: []policy.FSType{
-					"secret",
-					"emptyDir",
-				},
-			},
-		}
-	}
-	pspNode = func() policy.PodSecurityPolicy {
-		return policy.PodSecurityPolicy{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "lb-csi-node-sa",
-			},
-			Spec: policy.PodSecurityPolicySpec{
-				AllowedCapabilities: []corev1.Capability{"SYS_ADMIN"},
-				AllowedHostPaths: []policy.AllowedHostPath{
-					{PathPrefix: "/var/lib/kubelet"},
-					{PathPrefix: "/dev"},
-					{PathPrefix: "/lib/modules"},
-					{PathPrefix: "/var/lib/kubelet/*"},
-				},
-				HostNetwork:        true,
-				Privileged:         true,
-				FSGroup:            policy.FSGroupStrategyOptions{Rule: policy.FSGroupStrategyRunAsAny},
-				RunAsUser:          policy.RunAsUserStrategyOptions{Rule: policy.RunAsUserStrategyRunAsAny},
-				SELinux:            policy.SELinuxStrategyOptions{Rule: policy.SELinuxStrategyRunAsAny},
-				SupplementalGroups: policy.SupplementalGroupsStrategyOptions{Rule: policy.SupplementalGroupsStrategyRunAsAny},
-				Volumes: []policy.FSType{
-					"secret",
-					"emptyDir",
-					"hostPath",
-				},
-			},
-		}
+	// ServiceAccounts
+	ctrlServiceAccount = corev1.ServiceAccount{
```

Comment:
I think it was good that these snippets were wrapped in a getter function that returns a new object. This prevented modification of the global variable, which can happen easily when a developer forgets about it through the `controllerutil` funcs.

Reply:
I reintroduced the getter functions.

Reply:
Thanks. Also makes it much easier to review. :D